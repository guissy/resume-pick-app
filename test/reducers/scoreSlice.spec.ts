import { AnyAction } from 'redux';
import counterReducer, {
  updateNameScore,
  removeNameScore,
  clearNameScore,
  updateSearch,
} from '../../app/features/scoreSlice';

describe('reducers', () => {
  describe('scoreSlice', () => {
    it('should handle initial state', () => {
      expect(counterReducer(undefined, {} as AnyAction)).toMatchSnapshot();
    });

    it('should handle updateNameScore', () => {
      const newNameScore = {
        nameScore: [
          {
            name: 'name',
            path: 'path',
            score: 1,
            keywords: [],
            degree: '',
            links: [],
            school: '',
            text: '',
            level: '',
            levelValue: 0,
            levelSalary: 0,
            phone: '',
            workAge: 0,
            salary: '',
            sentiment: 0,
          },
        ],
        search: '',
      };
      expect(
        counterReducer(newNameScore, {
          type: updateNameScore,
          payload: newNameScore,
        })
      ).toMatchSnapshot();
    });
    it('should handle removeNameScore', () => {
      const newNameScore = {
        nameScore: [
          {
            name: 'name',
            path: 'path',
            score: 1,
            keywords: [],
            degree: '',
            links: [],
            school: '',
            text: '',
            level: '',
            levelValue: 0,
            levelSalary: 0,
            phone: '',
            workAge: 0,
            salary: '',
            sentiment: 0,
          },
        ],
        search: '',
      };
      expect(
        counterReducer(newNameScore, {
          type: removeNameScore,
          payload: newNameScore,
        })
      ).toMatchSnapshot();
    });
    it('should handle clearNameScore', () => {
      const newNameScore = {
        nameScore: [
          {
            name: 'name',
            path: 'path',
            score: 1,
            keywords: [],
            degree: '',
            links: [],
            school: '',
            text: '',
            level: '',
            levelSalary: 0,
            levelValue: 0,
            phone: '',
            workAge: 0,
            salary: '',
            sentiment: 0,
          },
        ],
        search: '',
      };
      expect(
        counterReducer(newNameScore, {
          type: clearNameScore,
        })
      ).toMatchSnapshot();
    });
    it('should handle updateSearch', () => {
      expect(
        counterReducer(
          { search: '', nameScore: [] },
          {
            type: updateSearch,
            payload: 'test',
          }
        )
      ).toMatchSnapshot();
    });
  });
});
