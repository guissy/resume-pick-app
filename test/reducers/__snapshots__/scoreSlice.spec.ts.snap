// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers scoreSlice should handle initial state 1`] = `
Object {
  "nameScore": Array [],
  "search": "",
}
`;

exports[`reducers scoreSlice should handle removeNameScore 1`] = `
Object {
  "nameScore": Array [],
  "search": "",
}
`;

exports[`reducers scoreSlice should handle updateNameScore 1`] = `
Object {
  "nameScore": Array [
    Object {
      "degree": "",
      "keywords": Array [],
      "level": "",
      "levelValue": 0,
      "links": Array [],
      "name": "name",
      "path": "path",
      "phone": "",
      "salary": "",
      "school": "",
      "score": 1,
      "sentiment": 0,
      "text": "",
      "workAge": 0,
    },
    Object {
      "nameScore": Array [
        Object {
          "degree": "",
          "keywords": Array [],
          "level": "",
          "levelValue": 0,
          "links": Array [],
          "name": "name",
          "path": "path",
          "phone": "",
          "salary": "",
          "school": "",
          "score": 1,
          "sentiment": 0,
          "text": "",
          "workAge": 0,
        },
      ],
      "search": "",
    },
  ],
  "search": "",
}
`;
